lib/data_structures/graphs/generic_graph.dart

```diff
@@ -23,17 +23,6 @@ class Node<T> {
   String toString() {
     return '{ value: $value, edges: [${edges.map((n) => n.value).join(',')}], hashCode: $hashCode }';
   }
-
-  @override
-  bool operator ==(Object other) {
-    return other is Node<T> &&
-        other.value == value &&
-        other.edges.length == edges.length &&
-        edges.every((edge) => other.edges.contains(edge));
```

Comment:
This was a bug that was causing infinite loops, it was also found that this equality override wasn't needed as it wasn't being used anywhere else before this PR (same goes for the hashCode).